Lib/concurrent/futures/thread.py

```diff
@@ -148,4 +150,5 @@ def shutdown(self, wait=True):
         if wait:
             for t in self._threads:
                 t.join()
+        self._threads.clear()
```

Comment:
> At least, this part should be under the if wait clause IMO.

I hesitated to not clear threads on shutdown(wait=False), but I don't see how clearing _theads can introduce any issue.

Anyway, I made the change ;-)